test/components/guideLineLayerTests.ts

```diff
@@ -0,0 +1,378 @@
+///<reference path="../testReference.ts" />
+
+describe("GuideLineLayer", () => {
+  it("scale()", () => {
+    let gll = new Plottable.Components.GuideLineLayer<Date>("vertical");
+    let timeScale = new Plottable.Scales.Time();
+    assert.strictEqual(gll.scale(timeScale), gll, "setter returns the calling GuideLineLayer");
+    assert.strictEqual(gll.scale(), timeScale, "getter returns the set Scale");
+  });
+
+  it("value()", () => {
+    let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+    let expectedValue = 5;
+    assert.isUndefined(gll.value(), "returns undefined before any value is set");
+    assert.strictEqual(gll.value(expectedValue), gll, "setter returns the calling GuideLineLayer");
+    assert.strictEqual(gll.value(), expectedValue, "getter returns the set value");
+  });
+
+  it("pixelPosition()", () => {
+    let gll = new Plottable.Components.GuideLineLayer<number>("vertical");
+    let expectedPosition = 5;
+    assert.isUndefined(gll.pixelPosition(), "returns undefined before any pixel position is set");
+    assert.strictEqual(gll.pixelPosition(expectedPosition), gll, "setter returns the calling GuideLineLayer");
+    assert.strictEqual(gll.pixelPosition(), expectedPosition, "getter returns the set pixel position");
+  });
```

Comment:
Do we allow `pixelPosition` and `value` to be set independently even with no backing scale?